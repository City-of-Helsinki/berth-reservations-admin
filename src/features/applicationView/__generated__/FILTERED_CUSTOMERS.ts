/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { OrganizationType } from "./../../../@types/__generated__/globalTypes";

// ====================================================
// GraphQL query operation: FILTERED_CUSTOMERS
// ====================================================

export interface FILTERED_CUSTOMERS_profiles_edges_node_primaryAddress {
  __typename: "AddressNode";
  address: string;
  postalCode: string;
  city: string;
}

export interface FILTERED_CUSTOMERS_profiles_edges_node_organization {
  __typename: "OrganizationNode";
  businessId: string;
  organizationType: OrganizationType;
}

export interface FILTERED_CUSTOMERS_profiles_edges_node_berthLeases_edges_node_berth_pier_properties_harbor_properties {
  __typename: "HarborProperties";
  name: string | null;
}

export interface FILTERED_CUSTOMERS_profiles_edges_node_berthLeases_edges_node_berth_pier_properties_harbor {
  __typename: "HarborNode";
  id: string;
  properties: FILTERED_CUSTOMERS_profiles_edges_node_berthLeases_edges_node_berth_pier_properties_harbor_properties | null;
}

export interface FILTERED_CUSTOMERS_profiles_edges_node_berthLeases_edges_node_berth_pier_properties {
  __typename: "PierProperties";
  identifier: string;
  harbor: FILTERED_CUSTOMERS_profiles_edges_node_berthLeases_edges_node_berth_pier_properties_harbor;
}

export interface FILTERED_CUSTOMERS_profiles_edges_node_berthLeases_edges_node_berth_pier {
  __typename: "PierNode";
  properties: FILTERED_CUSTOMERS_profiles_edges_node_berthLeases_edges_node_berth_pier_properties | null;
}

export interface FILTERED_CUSTOMERS_profiles_edges_node_berthLeases_edges_node_berth {
  __typename: "BerthNode";
  number: number;
  pier: FILTERED_CUSTOMERS_profiles_edges_node_berthLeases_edges_node_berth_pier;
}

export interface FILTERED_CUSTOMERS_profiles_edges_node_berthLeases_edges_node {
  __typename: "BerthLeaseNode";
  berth: FILTERED_CUSTOMERS_profiles_edges_node_berthLeases_edges_node_berth;
}

export interface FILTERED_CUSTOMERS_profiles_edges_node_berthLeases_edges {
  __typename: "BerthLeaseNodeEdge";
  node: FILTERED_CUSTOMERS_profiles_edges_node_berthLeases_edges_node | null;
}

export interface FILTERED_CUSTOMERS_profiles_edges_node_berthLeases {
  __typename: "BerthLeaseNodeConnection";
  edges: (FILTERED_CUSTOMERS_profiles_edges_node_berthLeases_edges | null)[];
}

export interface FILTERED_CUSTOMERS_profiles_edges_node_berthApplications_edges_node_berthSwitch {
  __typename: "BerthSwitchType";
  harborName: string;
}

export interface FILTERED_CUSTOMERS_profiles_edges_node_berthApplications_edges_node {
  __typename: "BerthApplicationNode";
  berthSwitch: FILTERED_CUSTOMERS_profiles_edges_node_berthApplications_edges_node_berthSwitch | null;
}

export interface FILTERED_CUSTOMERS_profiles_edges_node_berthApplications_edges {
  __typename: "BerthApplicationNodeEdge";
  node: FILTERED_CUSTOMERS_profiles_edges_node_berthApplications_edges_node | null;
}

export interface FILTERED_CUSTOMERS_profiles_edges_node_berthApplications {
  __typename: "BerthApplicationNodeConnection";
  edges: (FILTERED_CUSTOMERS_profiles_edges_node_berthApplications_edges | null)[];
}

export interface FILTERED_CUSTOMERS_profiles_edges_node {
  __typename: "ProfileNode";
  id: string;
  firstName: string;
  lastName: string;
  primaryAddress: FILTERED_CUSTOMERS_profiles_edges_node_primaryAddress | null;
  organization: FILTERED_CUSTOMERS_profiles_edges_node_organization | null;
  berthLeases: FILTERED_CUSTOMERS_profiles_edges_node_berthLeases | null;
  berthApplications: FILTERED_CUSTOMERS_profiles_edges_node_berthApplications | null;
}

export interface FILTERED_CUSTOMERS_profiles_edges {
  __typename: "ProfileNodeEdge";
  node: FILTERED_CUSTOMERS_profiles_edges_node | null;
}

export interface FILTERED_CUSTOMERS_profiles {
  __typename: "ProfileNodeConnection";
  count: number | null;
  edges: (FILTERED_CUSTOMERS_profiles_edges | null)[];
}

export interface FILTERED_CUSTOMERS {
  profiles: FILTERED_CUSTOMERS_profiles | null;
}

export interface FILTERED_CUSTOMERSVariables {
  first: number;
  after?: string | null;
  firstName?: string | null;
  lastName?: string | null;
  email?: string | null;
  address?: string | null;
  orderBy?: string | null;
}
