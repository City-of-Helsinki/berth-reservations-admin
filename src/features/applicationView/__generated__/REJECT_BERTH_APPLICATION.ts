/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { RejectBerthApplicationMutationInput } from "./../../../@types/__generated__/globalTypes";

// ====================================================
// GraphQL mutation operation: REJECT_BERTH_APPLICATION
// ====================================================

export interface REJECT_BERTH_APPLICATION_rejectBerthApplication {
  __typename: "RejectBerthApplicationMutationPayload";
  clientMutationId: string | null;
}

export interface REJECT_BERTH_APPLICATION {
  rejectBerthApplication: REJECT_BERTH_APPLICATION_rejectBerthApplication | null;
}

export interface REJECT_BERTH_APPLICATIONVariables {
  input: RejectBerthApplicationMutationInput;
}
