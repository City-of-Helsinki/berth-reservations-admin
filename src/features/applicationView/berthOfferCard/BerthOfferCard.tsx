import React from 'react';
import { IconTrash } from 'hds-react';
import { useTranslation } from 'react-i18next';
import { PureQueryOptions } from 'apollo-client';

import InternalLink from '../../../common/internalLink/InternalLink';
import { IconFence, IconPlug, IconStreetLight, IconWaterTap } from '../../../common/icons';
import Button from '../../../common/button/Button';
import InvoiceCard from '../../invoiceCard/InvoiceCardContainer';
import { PlaceProperty } from '../../invoiceCard/types';
import PlaceDetails from './PlaceDetails';
import { LeaseDetails } from './types';
import { ApplicationStatus } from '../../../@types/__generated__/globalTypes';

export interface BerthOfferCardProps {
  className?: string;
  applicationStatus: ApplicationStatus;
  leaseDetails: LeaseDetails;
  refetchQueries: PureQueryOptions[] | string[];
  handleDeleteLease: (id: string) => void;
}

const BerthOfferCard = ({
  className,
  applicationStatus,
  leaseDetails: {
    id,
    berthComment,
    berthDepth,
    berthIsAccessible,
    berthLength,
    berthMooringType,
    berthNum,
    berthWidth,
    customerEmail,
    electricity,
    gate,
    harborName,
    lighting,
    pierIdentifier,
    wasteCollection,
    water,
    order,
  },
  refetchQueries,
  handleDeleteLease,
}: BerthOfferCardProps) => {
  const { t } = useTranslation();

  const properties: PlaceProperty[] = [
    { prop: wasteCollection, key: 'wasteCollection', icon: IconTrash },
    { prop: electricity, key: 'electricity', icon: IconPlug },
    { prop: lighting, key: 'lighting', icon: IconStreetLight },
    { prop: gate, key: 'gate', icon: IconFence },
    { prop: water, key: 'water', icon: IconWaterTap },
  ];

  return (
    <>
      <InvoiceCard
        applicationStatus={applicationStatus}
        buttonsRight={
          <>
            <Button variant="supplementary" disabled>
              {t('offer.billing.showBill')}
            </Button>
            <Button variant="supplementary" disabled>
              {t('offer.billing.showContract')}
            </Button>
            <Button variant="secondary" theme="coat" onClick={() => handleDeleteLease(id)}>
              {t('offer.billing.removeOffer')}
            </Button>
          </>
        }
        className={className}
        customerEmail={customerEmail}
        order={order}
        placeDetails={
          <PlaceDetails
            berthMooringType={berthMooringType}
            berthWidth={berthWidth}
            berthLength={berthLength}
            berthDepth={berthDepth}
            berthIsAccessible={berthIsAccessible}
            berthComment={berthComment}
          />
        }
        placeName={
          <InternalLink to="/" underlined>
            {[harborName, pierIdentifier, berthNum].filter(Boolean).join(' ')}
          </InternalLink>
        }
        placeProperties={properties}
        placeType={t('common.terminology.berth').toUpperCase()}
        refetchQueries={refetchQueries}
        title={t('offer.title').toUpperCase()}
      />
    </>
  );
};

export default BerthOfferCard;
