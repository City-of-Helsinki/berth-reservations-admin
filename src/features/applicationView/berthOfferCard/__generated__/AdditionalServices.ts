/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { ProductServiceType, PriceUnits, PeriodType, AdditionalProductType } from "./../../../../@types/__generated__/globalTypes";

// ====================================================
// GraphQL fragment: AdditionalServices
// ====================================================

export interface AdditionalServices_additionalProducts_edges_node {
  __typename: "AdditionalProductNode";
  id: string;
  service: ProductServiceType;
  priceValue: any;
  priceUnit: PriceUnits;
  period: PeriodType;
  productType: AdditionalProductType;
}

export interface AdditionalServices_additionalProducts_edges {
  __typename: "AdditionalProductNodeEdge";
  node: AdditionalServices_additionalProducts_edges_node | null;
}

export interface AdditionalServices_additionalProducts {
  __typename: "AdditionalProductNodeConnection";
  edges: (AdditionalServices_additionalProducts_edges | null)[];
}

export interface AdditionalServices {
  __typename: "Query";
  additionalProducts: AdditionalServices_additionalProducts | null;
}
