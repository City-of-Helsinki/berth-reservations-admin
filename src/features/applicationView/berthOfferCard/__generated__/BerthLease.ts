/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { LeaseStatus, BerthMooringType, OrderStatus, ProductServiceType, AdditionalProductType } from "./../../../../@types/__generated__/globalTypes";

// ====================================================
// GraphQL fragment: BerthLease
// ====================================================

export interface BerthLease_berth_pier_properties_harbor_properties {
  __typename: "HarborProperties";
  name: string | null;
}

export interface BerthLease_berth_pier_properties_harbor {
  __typename: "HarborNode";
  id: string;
  properties: BerthLease_berth_pier_properties_harbor_properties | null;
}

export interface BerthLease_berth_pier_properties {
  __typename: "PierProperties";
  identifier: string;
  electricity: boolean;
  gate: boolean;
  lighting: boolean;
  mooring: boolean;
  wasteCollection: boolean;
  water: boolean;
  harbor: BerthLease_berth_pier_properties_harbor;
}

export interface BerthLease_berth_pier {
  __typename: "PierNode";
  id: string;
  properties: BerthLease_berth_pier_properties | null;
}

export interface BerthLease_berth {
  __typename: "BerthNode";
  id: string;
  depth: number | null;
  length: number;
  mooringType: BerthMooringType;
  width: number;
  comment: string;
  isAccessible: boolean | null;
  number: string;
  pier: BerthLease_berth_pier;
}

export interface BerthLease_customer_primaryEmail {
  __typename: "EmailNode";
  id: string;
  email: string;
}

export interface BerthLease_customer {
  __typename: "ProfileNode";
  id: string;
  primaryEmail: BerthLease_customer_primaryEmail | null;
}

export interface BerthLease_order_orderLines_edges_node_product {
  __typename: "AdditionalProductNode";
  id: string;
  service: ProductServiceType;
  productType: AdditionalProductType;
}

export interface BerthLease_order_orderLines_edges_node {
  __typename: "OrderLineNode";
  id: string;
  price: any;
  product: BerthLease_order_orderLines_edges_node_product | null;
}

export interface BerthLease_order_orderLines_edges {
  __typename: "OrderLineNodeEdge";
  node: BerthLease_order_orderLines_edges_node | null;
}

export interface BerthLease_order_orderLines {
  __typename: "OrderLineNodeConnection";
  edges: (BerthLease_order_orderLines_edges | null)[];
}

export interface BerthLease_order {
  __typename: "OrderNode";
  id: string;
  orderNumber: string;
  price: any;
  totalPrice: any;
  status: OrderStatus;
  orderLines: BerthLease_order_orderLines;
}

export interface BerthLease {
  __typename: "BerthLeaseNode";
  id: string;
  status: LeaseStatus;
  berth: BerthLease_berth;
  customer: BerthLease_customer;
  order: BerthLease_order | null;
}
