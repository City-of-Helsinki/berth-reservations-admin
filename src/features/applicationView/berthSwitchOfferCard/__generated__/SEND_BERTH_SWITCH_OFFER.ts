/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { SendBerthSwitchOfferMutationInput } from "./../../../../@types/__generated__/globalTypes";

// ====================================================
// GraphQL mutation operation: SEND_BERTH_SWITCH_OFFER
// ====================================================

export interface SEND_BERTH_SWITCH_OFFER_sendBerthSwitchOffer_failedOffers {
  __typename: "FailedOfferType";
  id: string;
  error: string | null;
}

export interface SEND_BERTH_SWITCH_OFFER_sendBerthSwitchOffer {
  __typename: "SendBerthSwitchOfferMutationPayload";
  failedOffers: (SEND_BERTH_SWITCH_OFFER_sendBerthSwitchOffer_failedOffers | null)[] | null;
  sentOffers: (string | null)[] | null;
}

export interface SEND_BERTH_SWITCH_OFFER {
  sendBerthSwitchOffer: SEND_BERTH_SWITCH_OFFER_sendBerthSwitchOffer | null;
}

export interface SEND_BERTH_SWITCH_OFFERVariables {
  input: SendBerthSwitchOfferMutationInput;
}
