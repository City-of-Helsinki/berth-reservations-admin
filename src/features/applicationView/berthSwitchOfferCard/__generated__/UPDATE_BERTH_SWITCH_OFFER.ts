/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { UpdateBerthSwitchOfferMutationInput } from "./../../../../@types/__generated__/globalTypes";

// ====================================================
// GraphQL mutation operation: UPDATE_BERTH_SWITCH_OFFER
// ====================================================

export interface UPDATE_BERTH_SWITCH_OFFER_updateBerthSwitchOffer {
  __typename: "UpdateBerthSwitchOfferMutationPayload";
  clientMutationId: string | null;
}

export interface UPDATE_BERTH_SWITCH_OFFER {
  updateBerthSwitchOffer: UPDATE_BERTH_SWITCH_OFFER_updateBerthSwitchOffer | null;
}

export interface UPDATE_BERTH_SWITCH_OFFERVariables {
  input: UpdateBerthSwitchOfferMutationInput;
}
