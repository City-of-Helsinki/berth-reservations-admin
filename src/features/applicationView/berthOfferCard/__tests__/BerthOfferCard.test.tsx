import React from 'react';
import { mount } from 'enzyme';
import { HashRouter } from 'react-router-dom';

import BerthOfferCard from '../BerthOfferCard';
import { BerthMooringType } from '../../../../@types/__generated__/globalTypes';

describe('BerthOfferCard', () => {
  const getWrapper = (
    props = {
      leaseDetails: {
        id: '123456',
        berthComment: 'Testi',
        berthDepth: 3,
        berthIsAccessible: true,
        berthLength: 6,
        berthMooringType: BerthMooringType.DINGHY_PLACE,
        berthNum: '1',
        berthWidth: 4,
        electricity: true,
        gate: true,
        harborName: 'Testisatama',
        lighting: true,
        pierIdentifier: 'Testilaituri',
        wasteCollection: true,
        water: true,
        order: null,
      },
      handleDeleteLease: jest.fn(),
      refetchQueries: [],
    }
  ) =>
    mount(
      <HashRouter>
        <BerthOfferCard {...props} />
      </HashRouter>
    );

  it('renders normally', () => {
    const wrapper = getWrapper();

    expect(wrapper.render()).toMatchSnapshot();
  });
});
