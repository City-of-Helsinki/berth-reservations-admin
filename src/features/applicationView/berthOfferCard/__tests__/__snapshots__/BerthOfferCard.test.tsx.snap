// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BerthOfferCard renders normally 1`] = `
Array [
  <div
    class="Select-module_root__Ka5uO invoiceActions"
  >
    <div
      class="Select-module_wrapper__1WQXs"
    >
      <button
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-labelledby="hds-select-1-label hds-select-1-toggle-button"
        aria-owns="hds-select-1-menu"
        class="Select-module_button__1aIsm Select-module_placeholder__21c0j"
        id="hds-select-1-toggle-button"
        type="button"
      >
        <span
          class="Select-module_buttonLabel__1fqu5"
        >
          Laskun muokkaus
        </span>
        <svg
          aria-hidden="true"
          class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik Select-module_angleIcon__2-9AD"
          role="img"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="none"
            fill-rule="evenodd"
          >
            <path
              d="M0 0h24v24H0z"
            />
            <path
              d="M12 13.5l5-5 1.5 1.5-6.5 6.5L5.5 10 7 8.5z"
              fill="currentColor"
            />
          </g>
        </svg>
      </button>
      <ul
        aria-labelledby="hds-select-1-label"
        class="Select-module_menu__1H2aU"
        id="hds-select-1-menu"
        role="listbox"
        style="max-height: 260px;"
        tabindex="-1"
      />
    </div>
  </div>,
  <div
    class="card offerCard"
  >
    <div
      class="cardHeader"
    >
      <span
        class="title"
      >
        TARJOUS
      </span>
    </div>
    <div
      class="cardBody"
    >
      <div
        class="grid cols3"
      >
        <article
          class="section"
        >
          <h4
            class="text standard h4 m title"
          >
            VENEPAIKKA
          </h4>
          <section
            class="body"
          >
            <article
              class="section"
            >
              <section
                class="body"
              >
                <a
                  class="internalLink brand underlined"
                  href="#/harbors/123"
                  title="Testisatama Testilaituri 1"
                >
                  Testisatama Testilaituri 1
                </a>
              </section>
            </article>
            <article
              class="section"
            >
              <section
                class="body"
              >
                <div
                  class="properties"
                >
                  <div
                    class="property property"
                  >
                    <div
                      class="iconWrapper standard outlined"
                    >
                      <svg
                        class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_m__3edUY icon_hds-icon--size-m__1mcHv"
                        role="img"
                        viewBox="0 0 24 24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <g
                          fill="none"
                          fill-rule="evenodd"
                        >
                          <path
                            d="M0 0h24v24H0z"
                          />
                          <path
                            d="M18 9H6l1.087 13h9.826L18 9zm-2.095 2l-.818 9H8.912l-.818-9h7.811zM12 2a2.97 2.97 0 012.995 2.82L15 5v1h3v2H6V6h3V5c0-1.688 1.331-3 3-3zm0 2c-.53 0-.938.365-.994.879L11 5v1h2V5c0-.577-.43-1-1-1z"
                            fill="currentColor"
                          />
                        </g>
                      </svg>
                    </div>
                    <span
                      class="text standard span s bold label"
                    >
                      Jätehuolto
                    </span>
                  </div>
                  <div
                    class="property property"
                  >
                    <div
                      class="iconWrapper standard outlined"
                    >
                      <svg
                        class="icon m"
                        height="100%"
                        viewBox="0 0 400 400"
                        width="100%"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M159.86 380.84h42v74.41h-42zM327.33 142.77H262.1v-81h-32v81h-98.47v-81h-32v81H34.39v32H67v48.82a111.59 111.59 0 0 0 21.6 65.91 119.83 119.83 0 0 0 43.63 36.86v42.86h97.24v-42.5a113.69 113.69 0 0 0 65.23-103.13v-48.82h32.62zm-64.62 80.82c0 34.94-21.88 66-54.46 77.24l-10.77 3.72v32.67h-33.24V305L154 301c-32.91-12.8-55-43.92-55-77.45v-48.78h163.7z"
                        />
                      </svg>
                    </div>
                    <span
                      class="text standard span s bold label"
                    >
                      Sähkö
                    </span>
                  </div>
                  <div
                    class="property property"
                  >
                    <div
                      class="iconWrapper standard outlined"
                    >
                      <svg
                        class="icon m"
                        height="100%"
                        viewBox="0 0 480 480"
                        width="100%"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M294.37 344.75h32v55.77h-32zM200.584 355.229l41.577-41.664 22.651 22.604-41.577 41.664zM362.922 336.221l22.65-22.604 41.577 41.664-22.65 22.604z"
                        />
                        <path
                          d="M321.32 143.85h-.95v-71.3h-126.7a93.65 93.65 0 0 0-67.59 28.32C108.36 119 98.6 143.43 98.6 169.57V461h95.26v-32H130.6V169.57c0-36.46 27.71-65 63.07-65h94.7v39.3h-.94c-64.88 0-117.66 49.82-117.66 111.07v16h73.43a60.61 60.61 0 0 0 20.8 30.33c13.82 11.08 32.38 17.18 52.26 17.18s38.44-6.1 52.26-17.18a60.61 60.61 0 0 0 20.78-30.35H439v-16c0-61.25-52.8-111.07-117.68-111.07zm-5.08 142.6c-16.55 0-30.36-6.1-37.67-15.53h75.35c-7.32 9.43-21.13 15.53-37.68 15.53zm75.3-47.53h-188c8-36 42.6-63.07 83.9-63.07h33.89c41.3 0 75.87 27.11 83.9 63.07z"
                        />
                      </svg>
                    </div>
                    <span
                      class="text standard span s bold label"
                    >
                      Valaistus
                    </span>
                  </div>
                  <div
                    class="property property"
                  >
                    <div
                      class="iconWrapper standard outlined"
                    >
                      <svg
                        class="icon m"
                        height="100%"
                        viewBox="0 0 480 480"
                        width="100%"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M410.18 85l-62 46.23v60.5h-25v-60.44L262.1 85l-62 46.23v60.5h-25v-60.44L114 85l-62 46.22v332.17h123.16V383.5h25v79.89h123.09V383.5h25v79.89h123.08v-332.1zm-178.06 62.27L262 125l29.3 22.18v44.52h-59.18zm59.13 76.45V351.5h-59.13V223.72zM143.16 431.39H84V147.27L113.87 125l29.29 22.18v44.52h-41v32h41v127.8h-41v32h41zm32-79.89V223.72h25V351.5zm116.09 79.89h-59.13V383.5h59.13zm32-79.89V223.72h25V351.5zm116.08 79.89h-59.12V383.5h39v-32h-39V223.72h39v-32h-39v-44.45L410 125l29.29 22.18z"
                        />
                      </svg>
                    </div>
                    <span
                      class="text standard span s bold label"
                    >
                      Portti
                    </span>
                  </div>
                  <div
                    class="property property"
                  >
                    <div
                      class="iconWrapper standard outlined"
                    >
                      <svg
                        class="icon m"
                        height="100%"
                        viewBox="0 0 480 480"
                        width="100%"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M154.61 313.9l-11.36-11.45-11.36 11.45a341.93 341.93 0 0 0-26.29 30.56c-18.95 25-28.16 45.16-28.16 61.73C77.44 425 84.87 442 98.37 454a68.35 68.35 0 0 0 89.76 0c13.5-12 20.94-29 20.94-47.81 0-16.57-9.21-36.76-28.16-61.73a341.09 341.09 0 0 0-26.3-30.56zm-11.36 124.9c-22.2 0-33.81-16.4-33.81-32.61 0-4.14 2.09-16.54 21.44-42.11 4.29-5.66 8.61-10.88 12.37-15.22 3.76 4.34 8.09 9.56 12.37 15.22 19.36 25.57 21.45 38 21.45 42.11 0 16.21-11.62 32.61-33.82 32.61zM341.91 158.33v-44h66.42v-32H243.49v32h66.42v44h-83.63c-57.36 0-104 45.76-104 102v21h42v-21c0-33.1 27.83-60 62-60h220.26v-42z"
                        />
                      </svg>
                    </div>
                    <span
                      class="text standard span s bold label"
                    >
                      Vesi
                    </span>
                  </div>
                </div>
              </section>
            </article>
          </section>
        </article>
        <article
          class="section"
        >
          <h4
            class="text standard h4 m title"
          >
            PAIKAN TIEDOT
          </h4>
          <section
            class="body"
          >
            <article
              class="section"
            >
              <section
                class="body"
              >
                <div
                  class="labelValuePair"
                >
                  <span
                    class="label standard"
                  >
                    Kiinnitystapa:
                  </span>
                  <span
                    class="value left"
                  >
                    Jollapaikka
                  </span>
                </div>
                <div
                  class="labelValuePair"
                >
                  <span
                    class="label standard"
                  >
                    Leveys:
                  </span>
                  <span
                    class="value left"
                  >
                    4,0 m
                  </span>
                </div>
                <div
                  class="labelValuePair"
                >
                  <span
                    class="label standard"
                  >
                    Pituus:
                  </span>
                  <span
                    class="value left"
                  >
                    6,0 m
                  </span>
                </div>
                <div
                  class="labelValuePair"
                >
                  <span
                    class="label standard"
                  >
                    Syvyys:
                  </span>
                  <span
                    class="value left"
                  >
                    3,0 m
                  </span>
                </div>
              </section>
            </article>
            <article
              class="section"
            >
              <section
                class="body"
              >
                <div
                  class="labelValuePair"
                >
                  <span
                    class="label standard"
                  >
                    Huoltotiedot (PLACEHOLDER):
                  </span>
                  <span
                    class="value left"
                  >
                    <a
                      class="internalLink brand"
                      href="#/"
                      title="123"
                    >
                      123
                    </a>
                    <br />
                    <a
                      class="internalLink brand"
                      href="#/"
                      title="456"
                    >
                      456
                    </a>
                  </span>
                </div>
                <div
                  class="labelValuePair"
                >
                  <span
                    class="label standard"
                  >
                    Huomiot:
                  </span>
                  <span
                    class="value left"
                  >
                    Testi
                  </span>
                </div>
              </section>
            </article>
          </section>
        </article>
      </div>
      <hr />
      <div
        class="buttonRow"
      >
        <div>
          <button
            aria-disabled="true"
            class="Button-module_button__1msFE button_hds-button__2A0je Button-module_primary__2LfKB button_hds-button--primary__2NVvO Button-module_theme-coat__2_3U7 button_hds-button--theme-coat__12cO0"
            disabled=""
            type="button"
          >
            <span
              class="Button-module_label__a4np1 button_hds-button__label__2EQa-"
            >
              Hyväksy ja lähetä tarjous
            </span>
          </button>
        </div>
        <div>
          <button
            aria-disabled="true"
            class="Button-module_button__1msFE button_hds-button__2A0je Button-module_supplementary__3YKiS button_hds-button--supplementary__GcHcV Button-module_theme-coat__2_3U7 button_hds-button--theme-coat__12cO0"
            disabled=""
            type="button"
          >
            <div
              aria-hidden="true"
              class="Button-module_icon__O-h7R button_hds-icon__17j8Z"
            >
              <svg
                class="icon s"
                height="100%"
                viewBox="0 0 24 24"
                width="100%"
                xmlns="http://www.w3.org/2000/svg"
              >
                <title>
                  icons/document/24
                </title>
                <g
                  fill="none"
                  fill-rule="evenodd"
                  id="icons/document/24"
                  stroke="none"
                  stroke-width="1"
                >
                  <path
                    d="M21,1 L21,23 L3,23 L3,9 L4.741,9 L4.7419366,21.3500004 L19.2580646,21.3500004 L19.2580646,2.65 L12,2.65 L12,1 L21,1 Z M17.6,18 L17.6,19.7 L6,19.7 L6,18 L17.6,18 Z M10,15 L10,16.7 L6,16.7 L6,15 L10,15 Z M4.26243563,3.08071012 L8.70639967,7.52507832 L11.7375646,4.49492357 L12.9497476,5.70710662 L9.9183807,8.73705934 L9.91928988,8.73756437 L8.70710678,9.94974747 L3.05025253,4.29289322 L4.26243563,3.08071012 Z"
                    fill="currentColor"
                    id="aggreements_icon"
                    transform="translate(12.000000, 12.000000) scale(-1, 1) translate(-12.000000, -12.000000) "
                  />
                </g>
              </svg>
            </div>
            <span
              class="Button-module_label__a4np1 button_hds-button__label__2EQa-"
            >
              Näytä lasku
            </span>
          </button>
          <button
            aria-disabled="true"
            class="Button-module_button__1msFE button_hds-button__2A0je Button-module_supplementary__3YKiS button_hds-button--supplementary__GcHcV Button-module_theme-coat__2_3U7 button_hds-button--theme-coat__12cO0"
            disabled=""
            type="button"
          >
            <div
              aria-hidden="true"
              class="Button-module_icon__O-h7R button_hds-icon__17j8Z"
            >
              <svg
                class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
                role="img"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  fill="none"
                  fill-rule="evenodd"
                >
                  <rect
                    height="24"
                    width="24"
                  />
                  <path
                    d="M15,2 L20,7 L20,22 L4,22 L4,2 L15,2 Z M14,4 L14,8 L18,8 L14,4 Z M14.0082581,4 L18,8 L18,20 L6,20 L6,4 L14.0082581,4 Z M17,16 L7,16 L7,18 L17,18 L17,16 Z M17,12 L7,12 L7,14 L17,14 L17,12 Z M13,8 L7,8 L7,10 L13,10 L13,8 Z"
                    fill="currentColor"
                  />
                </g>
              </svg>
            </div>
            <span
              class="Button-module_label__a4np1 button_hds-button__label__2EQa-"
            >
              Näytä sopimus
            </span>
          </button>
          <button
            class="Button-module_button__1msFE button_hds-button__2A0je Button-module_secondary__1nABp button_hds-button--secondary__1NOWS Button-module_theme-coat__2_3U7 button_hds-button--theme-coat__12cO0"
            type="button"
          >
            <span
              class="Button-module_label__a4np1 button_hds-button__label__2EQa-"
            >
              Poista tarjous
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>,
]
`;
